config/config-api/test/com/thoughtworks/go/config/TemplatesConfigTest.java

```diff
@@ -114,7 +115,41 @@ public void shouldReturnFalseIfUserCannotViewAndEditAtLeastOneTemplate() throws
         templateList.add(PipelineTemplateConfigMother.createTemplate("template100", new Authorization(new AdminsConfig(new AdminUser(templateAdmin))), StageConfigMother.manualStage("stage-name")));
         TemplatesConfig templates = new TemplatesConfig(templateList.toArray(new PipelineTemplateConfig[0]));
 
-        assertThat(templates.canViewAndEditTemplate(nonTemplateAdmin), is(false));
+        assertThat(templates.canViewAndEditTemplate(nonTemplateAdmin, null), is(false));
+    }
+
+    @Test
+    public void shouldReturnTrueIfUserWithinARoleCanViewAndEditTemplates() {
+        ArrayList<PipelineTemplateConfig> templateList = new ArrayList<PipelineTemplateConfig>();
+        for (int i = 0; i < 100; i++) {
+            templateList.add(PipelineTemplateConfigMother.createTemplate("template" + i));
+        }
+        CaseInsensitiveString templateAdmin = new CaseInsensitiveString("template-admin");
+        Role securityConfigRole = new Role(new CaseInsensitiveString("role1"), new RoleUser(templateAdmin));
+        AdminRole templateAdminRole = new AdminRole(securityConfigRole);
+        templateList.add(PipelineTemplateConfigMother.createTemplate("templateName", new Authorization(new AdminsConfig(templateAdminRole)), StageConfigMother.manualStage("stage-name")));
+        TemplatesConfig templates = new TemplatesConfig(templateList.toArray(new PipelineTemplateConfig[0]));
+        List<Role> roles = new ArrayList<>();
+        roles.add(securityConfigRole);
+
+        assertThat(templates.canViewAndEditTemplate(templateAdmin, roles), is(true));
```

Comment:
I found it a little hard to understand this test. 101 templates are added. It seems like only one (the last one) has some authorization added to to. The stage name param seems irrelevant and could be renamed to "dont-care" or something to show that.

Then, when you ask whether either of `templateAdmin` or `roles` can access the template, it says "yes". To make it more clear, you can do this, I feel:

```
assertThat(templates.canViewAndEditTemplate(someRandomNormalUser, roleWithTemplateAdmin), is(true));
```